### PR TITLE
Fixes wrong visitor group assumed on multistore

### LINKED ISSUE #10849: Wrong visitor group assumed on multistore
**Describe the bug**

Every time the `FrontController::getCurrentCustomerGroups()` method is called and returns an empty array, the customer group is assumed to be `1` by default. But In Group admin page, you've got the choice to define visitor group to a different id than `1`.

We should be using `Group::getCurrent()->id` instead.

**To Reproduce**

* Create a new group and define it as the visitor group.
* Create a category only visible by that group.
* Link a product to this category.
* Make a search call on this product name in frontend.
* The product will not show.

**Additionnal information**
PrestaShop version: all versions
PHP version: all versions

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Default group should not be id `1` but the visitor group set by the shop owner.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #10849 
| How to test?  | Follow steps in above ticket

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10851)

### CODE REVIEW

File: classes/Product.php [hunk 1]
```diff
@@ -2593,8 +2593,8 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
         if (Group::isFeatureActive()) {
             $groups = FrontController::getCurrentCustomerGroups();
             $sql_groups = ' AND EXISTS(SELECT 1 FROM `' . _DB_PREFIX_ . 'category_product` cp
-                JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '= ' . (int) Configuration::get('PS_UNIDENTIFIED_GROUP')) . ')
-                WHERE cp.`id_product` = p.`id_product`)';
+				JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Group::getCurrent()->id) . ')
```
Comment (PierreRambaud) [inline classes/Product.php]: No tabs please :)

File: classes/Product.php [hunk 2]
```diff
@@ -2649,8 +2649,8 @@ public static function getNewProducts($id_lang, $page_number = 0, $nb_products =
         if (Group::isFeatureActive()) {
             $groups = FrontController::getCurrentCustomerGroups();
             $sql->where('EXISTS(SELECT 1 FROM `' . _DB_PREFIX_ . 'category_product` cp
-                JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Configuration::get('PS_UNIDENTIFIED_GROUP')) . ')
-                WHERE cp.`id_product` = p.`id_product`)');
+				JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Group::getCurrent()->id) . ')
```
Comment (PierreRambaud) [inline classes/Product.php]: No tabs please :)

File: classes/Product.php [hunk 3]
```diff
@@ -2738,8 +2738,8 @@ public static function getRandomSpecial($id_lang, $beginning = false, $ending =
 
             $groups = FrontController::getCurrentCustomerGroups();
             $sql_groups = ' AND EXISTS(SELECT 1 FROM `' . _DB_PREFIX_ . 'category_product` cp
-                JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Configuration::get('PS_UNIDENTIFIED_GROUP')) . ')
-                WHERE cp.`id_product` = p.`id_product`)';
+				JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Group::getCurrent()->id) . ')
```
Comment (PierreRambaud) [inline classes/Product.php]: No tabs please :)

**Comment by matthieu-rolland** [comment]
This is a remake of this PR:  https://github.com/PrestaShop/PrestaShop/pull/10851

I fixed the conflicts, and made a new PR because I didn't want to push on @Flowster's own develop branch, since he made the fix on his develop branch.

File: classes/Product.php [hunk 4]
```diff
@@ -2863,8 +2863,8 @@ public static function getPricesDrop(
         if (Group::isFeatureActive()) {
             $groups = FrontController::getCurrentCustomerGroups();
             $sql_groups = ' AND EXISTS(SELECT 1 FROM `' . _DB_PREFIX_ . 'category_product` cp
-                JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Configuration::get('PS_UNIDENTIFIED_GROUP')) . ')
-                WHERE cp.`id_product` = p.`id_product`)';
+				JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Group::getCurrent()->id) . ')
```
Comment (PierreRambaud) [inline classes/Product.php]: No tabs please :)

File: classes/ProductSale.php [hunk 5]
```diff
@@ -138,7 +138,7 @@ public static function getBestSales($idLang, $pageNumber = 0, $nbProducts = 10,
         if (Group::isFeatureActive()) {
             $groups = FrontController::getCurrentCustomerGroups();
             $sql .= ' AND EXISTS(SELECT 1 FROM `' . _DB_PREFIX_ . 'category_product` cp
-					JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Configuration::get('PS_UNIDENTIFIED_GROUP')) . ')
+					JOIN `' . _DB_PREFIX_ . 'category_group` cg ON (cp.id_category = cg.id_category AND cg.`id_group` ' . (count($groups) ? 'IN (' . implode(',', $groups) . ')' : '=' . (int) Group::getCurrent()->id) . ')
```
Comment (PierreRambaud) [inline classes/ProductSale.php]: No tabs please :)

Comment (matthieu-rolland) [reply classes/ProductSale.php]: done !

File: classes/Product.php (hunk 4 above)
Comment (matthieu-rolland) [reply classes/Product.php]: done !

File: classes/Product.php (hunk 3 above)
Comment (matthieu-rolland) [reply classes/Product.php]: done !

File: classes/Product.php (hunk 2 above)
Comment (matthieu-rolland) [reply classes/Product.php]: done !

File: classes/Product.php (hunk 1 above)
Comment (matthieu-rolland) [reply classes/Product.php]: done !

**Review by matks — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**